testsuite/classlibrary/TestUGen_RTAlloc.sc

```diff
@@ -72,16 +72,24 @@ TestUGen_RTAlloc : UnitTest {
 		var output = this.awaitSynthOutput {
 			DelayN.ar(DC.ar(1), this.memSizeSeconds * 2)
 		};
-		this.assertEquals(output[0], 0.0,
-			"a failed RTAlloc should clear unit outputs")
+		if (output.isNil) {
+			this.assert(false, "test should complete")
+		} {
+			this.assertEquals(output[0], 0.0,
+				"a failed RTAlloc should clear unit outputs")
+		}
```

Comment:
Suggestion: an idea for this pattern of if/else that occurs multiple times—it can be made more concise throughout (untested):
```supercollider
this.assert(output.notNil, "test should complete (timeout)", onFailure: {});
this.assertEquals(output[0], 0.0, "a failed RTAlloc should clear unit outputs");
```
The docs indicate that providing a non nil value to `onFailure` will halt the test. So the method can return immediately after timing out.